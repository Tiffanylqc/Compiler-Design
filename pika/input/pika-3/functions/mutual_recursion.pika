func fib1<int n>->int {
	if (n == 0 || n == 1) {
		return 1.
	} else {
		return fib2(n - 1) + fib2(n - 2).
	}
}

func fib2<int n>->int {
	if (n == 0 || n == 1) {
		return 1.
	} else {
		return fib1(n - 1) + fib1(n - 2).
	}
}

exec {
	print "Fib(", 10, ") is ", fib2(10), _n_.
}