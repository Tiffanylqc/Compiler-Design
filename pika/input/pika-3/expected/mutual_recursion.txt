Fib(10) is 89
